services/src/pro/handlers/drone_mapping.rs

```diff
@@ -396,219 +394,226 @@ mod tests {
-                "POST",
-                "/task/new/upload/9c64ff33-b6d7-4ff9-b63c-aaba37dfb4b7",
-            ))
-            .times(2)
-            .respond_with(json_encoded(json!({}))),
-        );
-
-        mock_nodeodm.expect(
-            Expectation::matching(request::method_path(
-                "POST",
-                "/task/new/commit/9c64ff33-b6d7-4ff9-b63c-aaba37dfb4b7",
-            ))
-            .respond_with(json_encoded(json!({}))),
-        );
-
-        // manipulate config to use the mock nodeodm server
-        config::set_config("odm.endpoint", mock_nodeodm.url_str("/")).unwrap();
-
-        let app_ctx = ProInMemoryContext::test_default();
-        let session = create_session_helper(&app_ctx).await;
-        let ctx = app_ctx.session_context(session.clone());
-
-        // file upload into geo engine
-        // TODO: properly upload the data using the handler once this is possible in a test (after migration to actix)
-        let upload_id = UploadId::new();
-        let upload_dir = upload_id.root_path()?;
-        fs::create_dir_all(&upload_dir).await.context(error::Io)?;
-
-        test_data.uploads.push(upload_id);
-
-        // copy the test data into upload directory
-        let mut drone_images_dir = fs::read_dir(test_data!("pro/drone_mapping/drone_images"))
-            .await
-            .context(error::Io)?;
+        with_pro_temp_context(|app_ctx, _| async move {
+            let mut test_data = TestDataUploads::default(); // remember created folder and remove them on drop
+
+            let mock_nodeodm = Server::run();
+            mock_nodeodm.expect(
+                Expectation::matching(request::method_path("POST", "/task/new/init")).respond_with(
+                    json_encoded(json!({
+                        "uuid": "9c64ff33-b6d7-4ff9-b63c-aaba37dfb4b7"
+                    })),
+                ),
+            );
+
+            mock_nodeodm.expect(
+                Expectation::matching(request::method_path(
+                    "POST",
+                    "/task/new/upload/9c64ff33-b6d7-4ff9-b63c-aaba37dfb4b7",
+                ))
+                .times(2)
+                .respond_with(json_encoded(json!({}))),
+            );
+
+            mock_nodeodm.expect(
+                Expectation::matching(request::method_path(
+                    "POST",
+                    "/task/new/commit/9c64ff33-b6d7-4ff9-b63c-aaba37dfb4b7",
+                ))
+                .respond_with(json_encoded(json!({}))),
+            );
+
+            // manipulate config to use the mock nodeodm server
+            config::set_config("odm.endpoint", mock_nodeodm.url_str("/")).unwrap();
+
+            let session = create_session_helper(&app_ctx).await;
+            let ctx = app_ctx.session_context(session.clone());
+
+            // file upload into geo engine
+            // TODO: properly upload the data using the handler once this is possible in a test (after migration to actix)
```

Comment:
is this easy to do?

we did the migration to actix some year(s) ago

Reply:
I think this is possible now, yes. But we wanted to get rid of this handler entirely, or not?

Reply:
:+1:

Reply:
if it is easy to do, yes

Reply:
4bf61de4aadd39d243da6547f8505033e190573a